dz13/src/app.js

```diff
@@ -0,0 +1,126 @@
+function entriesArray(array) {
+  return array.flat(Infinity)
+}
+
+// function setKeysAndObjectInfo(user) {
+//   const entriesUser = Object.entries(user)
+//   const keyUser = []
+//   for (const key of entriesUser) {
+//     key.pop()
+//     keyUser.push(key)
+//   }
+//   console.log(keyUser.flat())
+// }
+function setKeysAndObjectInfo(user, type) {
+  if (type === 'key') {
+    const newUser = Object.keys(user)
+    return newUser
+  }
+  const newUser = Object.values(user)
+  return newUser
+}
+
+function logInCheck(cart) {
+  const newCart = Object.entries(cart)
+  for (const logCart of newCart) {
+    console.log(`${logCart[0]}: ${logCart[1]} шт.`)
+  }
+}
+
+function mergeSettings(defaultSettings, userPreferences) {
+  const resultSettings = { ...defaultSettings, ...userPreferences }
+  return resultSettings
```

Comment:
```suggestion
  return { ...defaultSettings, ...userPreferences }
```

создавать переменную не нужно